src/common/ui/setting-text.vue

```diff
@@ -32,11 +32,11 @@ export default {
     };
   },
   created() {
-    const transform = this.json
+    const handle = this.json
       ? (value => JSON.stringify(value, null, '  '))
       // XXX compatible with old data format
       : (value => (Array.isArray(value) ? value.join('\n') : value || ''));
-    this.revoke = hookSetting(this.name, { target: this, prop: 'value', transform });
+    this.revoke = hookSetting(this.name, val => { this.value = handle(val); }, true);
```

Comment:
maybe it'd be better to use `await options.ready` instead of `true` like we do in other places where hookSettings is used?

Reply:
Yep, did so.